build.gradle

```diff
@@ -382,6 +386,7 @@ sourceSets {
 springBoot {
     buildInfo()
     mainClass = 'org.pih.warehouse.Application'
+    executable = true
```

Comment:
Added this so that I could test the database migration with the application being deployed with standalone war i.e. embedded tomcat. 
```
cd build/libs
./openboxes.war
```
We should also test to make sure the everything works properly when deployed to an actual Tomcat server.

Reply:
It seems like this gives us the benefit of the creating an executable WAR file without the downside of dependency issues on the deployment side (i.e. embedded tomcat classes conflicting with the container classes installed on the server). 


The WAR files are about the same size
```
-rwxrw-r--  1 jmiranda jmiranda 187060158 May 15 12:09 openboxes-standalone.war
-rw-rw-r--  1 jmiranda jmiranda 187051977 May 15 12:12 openboxes.war
```
They both have the same number of files (and probably the same JARs)
```
[jmiranda@jmiranda-ThinkPad-W540 libs (OBPIH-6354-resolve-upgrade-version)]$ jar -tvf openboxes.war | wc -l
16141
[jmiranda@jmiranda-ThinkPad-W540 libs (OBPIH-6354-resolve-upgrade-version)]$ jar -tvf openboxes-standalone.war | wc -l
16141
```

The only difference seems to be in the way the files are "executable".

The "executable" WAR seems to have a bash script which can be seen if you `head` the file (see 

```
[jmiranda@jmiranda-ThinkPad-W540 libs (OBPIH-6354-resolve-upgrade-version)]$ head -10 openboxes-standalone.war 
#!/bin/bash
#
#    .   ____          _            __ _ _
#   /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
#  ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
#   \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
#    '  |____| .__|_| |_|_| |_\__, | / / / /
#   =========|_|==============|___/=/_/_/_/
#   :: Spring Boot Startup Script ::
#
```
According to the Spring Boot docs 

> Fully executable jars work by embedding an extra script at the front of the file. Currently, some tools do not accept this format, so you may not always be able to use this technique.

Source: https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/deployment-install.html

This means that the executable WAR file can be executed like a binary file.
```
jmiranda@jmiranda-ThinkPad-W540 libs (OBPIH-6354-resolve-upgrade-version)]$ ./openboxes-standalone.war 
12:14:19,807 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
12:14:19,807 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/home/jmiranda/git/openboxes-grails-migration/build/libs/openboxes-standalone.war!/WEB-INF/classes!/logback.xml]
12:14:19,825 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@7d417077 - URL [jar:file:/home/jmiranda/git/openboxes-grails-migration/build/libs/openboxes-standalone.war!/WEB-INF/classes!/logback.xml] is not of type file
```
while the non-executable WAR can requires you to execute the WAR using `java -jar`
```
[jmiranda@jmiranda-ThinkPad-W540 libs (OBPIH-6354-resolve-upgrade-version)]$ java -jar openboxes.war
12:15:55,018 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
12:15:55,018 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [jar:file:/home/jmiranda/git/openboxes-grails-migration/build/libs/openboxes.war!/WEB-INF/classes!/logback.xml]
12:15:55,038 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@7d417077 - URL [jar:file:/home/jmiranda/git/openboxes-grails-migration/build/libs/openboxes.war!/WEB-INF/classes!/logback.xml] is not of type file
12:15:55,224 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
```

So they are essentially the same, execute for that wrapper bash script. Do you think it's ok to keep it in or would you feel more comfortable if this was moved to a discovery ticket?

cc @EWaterman @awalkowiak @kchelstowski @alannadolny @drodzewicz

Reply:
as long as you can still run it via java -jar (which it seems like you can), I don't have any issues with making it executable